lib/virt.py

```diff
@@ -382,23 +385,26 @@ def install(self, location=None, kickstart=None, ks_verbatim=False):
             with open(f'{self.ssh_keyfile_path}.pub') as f:
                 pubkey = f.read().rstrip()
             kickstart.add_authorized_key(pubkey)
-            kickstart.add_install_only_repo('rpmpack', f'http://{NETWORK_HOST}:{http_port}/repo')
+            kickstart.add_install_only_repo(
+                'contest-rpmpack',
+                f'http://{NETWORK_HOST}:{http_port}/repo',
+            )
             kickstart.add_packages([util.RPMPACK_NAME])
 
         disk_path = f'{GUEST_IMG_DIR}/{self.name}.img'
         disk_format = 'raw'
         cpus = os.cpu_count() or 1
 
-        pack = util.RpmPack()
-        pack.post.append(self.SETUP)
-        pack.requires += self.SETUP_REQUIRES
-
-        srv = util.BackgroundHTTPServer(NETWORK_HOST, http_port)
-
         with contextlib.ExitStack() as stack:
+            pack = util.RpmPack()
+            pack.post.append(self.SETUP)
+            pack.requires += self.SETUP_REQUIRES
```

Comment:
Why moving it to the context?

Reply:
Because it seemed clearer and kept related lines together.

Adding some comments maybe explains it better:
```python
        with contextlib.ExitStack() as stack:
            # this is all RpmPack
            pack = util.RpmPack()
            pack.post.append(self.SETUP)
            pack.requires += self.SETUP_REQUIRES
            repo = stack.enter_context(pack.build_as_repo())

            # this is all HTTP server
            srv = util.BackgroundHTTPServer(NETWORK_HOST, http_port)
            srv.add_file(repo, 'repo')
            stack.enter_context(srv)

            # this is all Kickstart
            ksfile = stack.enter_context(kickstart.to_tmpfile())
```
Yes, it means that if `BackgroundHTTPServer` fails to initialize an instance, `pack.build_as_repo()` has wasted 1 second doing extra work that will be undone, but that should never happen in practice - instance initialization is not listening on a socket, which would be done after repo building anyway in the old code:
```python
        with contextlib.ExitStack() as stack:
            repo = stack.enter_context(pack.build_as_repo())
            srv.add_file(repo, 'repo')
            stack.enter_context(srv)
            ksfile = stack.enter_context(kickstart.to_tmpfile())
```

Reply:
Alright, I'm fine with that.

The idea, or rather how I write context (and it might be wrong that I made up it myself), is to put as less as possible to context and everything what doesn't need to be in it should stay outside. But looking at it now (lines together with comments), your approach is more sexy.

Reply:
FTR; I agree with you in general, don't put things that don't need to be in the context block, and that's how it should be for single-object contexts, it's just that in this case, I think trading that principle for the extra readability is worth it.